packages/remix/src/ssr/types.ts

```diff
@@ -31,11 +31,14 @@ export type RootAuthLoaderCallback<Options extends RootAuthLoaderOptions> = (
   args: LoaderFunctionArgsWithAuth<Options>,
 ) => RootAuthLoaderCallbackReturn;
 
-export type RootAuthLoaderCallbackReturn =
-  | Promise<Response>
-  | Response
-  | Promise<Record<string, unknown>>
-  | Record<string, unknown>;
+type ObjectLike = Record<string, unknown> | null;
```

Comment:
I think this provides some clarity around the intent of the type, along with the comment below.